File: BAEKJOON/Python/6000/6300/6322.py
"""
Title : 직각 삼각형의 두 변
Link : https://www.acmicpc.net/problem/6322
"""

from sys import stdin

input = stdin.readline


if __name__ == "__main__":
    triangle_num = 1
    ans = []
    while True:
        a, b, c = map(int, input().split())
        if a == b == c == 0:
            break
        tmp = f"Triangle #{triangle_num}\n"
        triangle_num += 1
        if c == -1:
            tmp += "c = %.3f\n" % ((a**2 + b**2) ** 0.5)
        elif a == -1:
            if b >= c:
                tmp += "Impossible.\n"
            else:
                tmp += "a = %.3f\n" % ((c**2 - b**2) ** 0.5)
        elif b == -1:
            if a >= c:
                tmp += "Impossible.\n"
            else:
                tmp += "b = %.3f\n" % ((c**2 - a**2) ** 0.5)
        ans.append(tmp)
    print("\n".join(ans))

File: BAEKJOON/Golang/6000/6300/6322.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
	"strconv"
	"strings"
)

var sc = bufio.NewScanner(os.Stdin)
var wr = bufio.NewWriter(os.Stdout)

func nextInts() [3]int {
	sc.Scan()
	text := sc.Text()
	tmp := strings.Split(text, " ")
	var nums [3]int
	for i, x := range tmp {
		n, _ := strconv.Atoi(x)
		nums[i] = n
	}
	return nums
}

func IntPow(x int) float64 {
	return float64(x * x)
}

func main() {
	sc.Split(bufio.ScanLines)
	defer wr.Flush()

	end := [3]int{0, 0, 0}
	triangleNum := 1
	for {
		nums := nextInts()
		if nums == end {
			break
		}
		fmt.Fprintf(wr, "Triangle #%d\n", triangleNum)
		triangleNum++
		if nums[2] == -1 {
			fmt.Fprintf(wr, "c = %.3f\n", math.Sqrt(IntPow(nums[0])+IntPow(nums[1])))
		} else if nums[0] == -1 {
			if nums[1] >= nums[2] {
				fmt.Fprint(wr, "Impossible.\n")
			} else {
				fmt.Fprintf(wr, "a = %.3f\n", math.Sqrt(IntPow(nums[2])-IntPow(nums[1])))
			}
		} else if nums[1] == -1 {
			if nums[0] >= nums[2] {
				fmt.Fprint(wr, "Impossible.\n")
			} else {
				fmt.Fprintf(wr, "b = %.3f\n", math.Sqrt(IntPow(nums[2])-IntPow(nums[0])))
			}
		}
		fmt.Fprint(wr, "\n")
	}
}
